click/termui.py

```diff
@@ -171,7 +175,7 @@ def confirm(text, default=False, abort=False, prompt_suffix=': ',
             rv = True
         elif value in ('n', 'no'):
             rv = False
-        elif value == '':
+        elif default is not None and value == '':
```

Comment:
@yurihs Changing 164/165 above to the following might make sense:
```python
    prompt = click.termui._build_prompt(
        text, prompt_suffix, show_default,
        'y/n' if default is None else default and 'Y/n' or 'y/N',
    )
```

Reply:
Thanks for reviewing! Yeah, I hadn't noticed that part. I've implemented your change.

I totally mixed up the functions in the changelog entry 😅

I think everything is OK now.